Symfony/CS/Fixer/Contrib/Php4ConstructorFixer.php

```diff
@@ -0,0 +1,183 @@
+                // That's pretty bad, delete the new one
+                for ($i = $p5['startIdx']; $i <= $p5['endIdx']; ++$i) {
+                    $tokens[$i]->clear();
+                }
+                // And rename the old one
+                $tokens[$p4['nameIdx']]->setContent('__construct');
+            }
+        }
+
+        // Check calls to the parent constructor
+        foreach ($tokens->findGivenKind(T_EXTENDS) as $index => $token) {
+            $parIndex = $tokens->getNextMeaningfulToken($index);
+            $parentClass = $tokens[$parIndex]->getContent();
+
+            // using parent::ParentClassName()
+            $currIndex = $parIndex;
+            while ($currIndex = $tokens->getNextTokenOfKind($currIndex, array(array(T_PAAMAYIM_NEKUDOTAYIM, "::")))) {
+                $prevIndex = $tokens->getPrevMeaningfulToken($currIndex);
+                if ($tokens[$prevIndex]->equals(array(T_STRING, 'parent'))) {
+                    $currIndex = $tokens->getNextMeaningfulToken($currIndex);
+                    if (!strcasecmp($tokens[$currIndex]->getContent(), $parentClass)) {
+                        // Found parent
+                        $tokens[$currIndex]->setContent('__construct');
+                        break;
+                    }
+                }
+            }
+
+            // using $this->ParentClassName()
+            $currIndex = $parIndex;
+            while ($currIndex = $tokens->getNextTokenOfKind($currIndex, array(array(T_OBJECT_OPERATOR, "->")))) {
+                $prevIndex = $tokens->getPrevMeaningfulToken($currIndex);
+                if ($tokens[$prevIndex]->equals(array(T_VARIABLE, '$this'))) {
+                    $operatorIndex = $currIndex;
+                    $currIndex = $tokens->getNextMeaningfulToken($currIndex);
+                    if (!strcasecmp($tokens[$currIndex]->getContent(), $parentClass)) {
+                        // Found parent
+                        $tokens[$prevIndex]->setContent('parent');
+                        $tokens[$operatorIndex]->setContent('::');
+                        $tokens[$currIndex]->setContent('__construct');
+
+                        break;
+                    }
+                }
+            }
+        }
+
+        return $tokens->generateCode();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getName()
+    {
+        return 'php4_constructor';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDescription()
+    {
+        return 'Convert PHP4-style constructors to __construct. Warning! This could change code behavior.';
+    }
+
+    /**
+     * Find a function matching a given name within certain bounds.
+     *
+     * @param  Tokens     $tokens Tokens instance
+     * @param  string     $name   Function/Method name
+     * @param  int        $start  Start index
+     * @param  int        $end    End index
+     * @return array|null {
```

Comment:
add blank line between `@param` and `@return`: namespace Foo
{
    class FooClass
    {
        public function FooClass() {} // leave alone!
    }
}

class Bar
{
    public function Bar() {} // need to be fixed !
}